### ISSUE TITLE
Create C  bindings, to then be able to do Python bindings

### ISSUE THREAD

**Issue by tmontaigu** [tmontaigu/laz-rs#2]
But not before at least version 3 of the compression & decompression are done and well tested

**Comment by hugoledoux** [comment]
Just see this and I wonder why going to C first to go to Python. Why not Python directly? I used this:
https://github.com/PyO3/maturin
and it’s great, but I can’t compare much, was my first attempt.

Code there if that helps:
https://github.com/hugoledoux/startin_python

When the LAZ rust is finished, or in a usable state, I’d love to use it in my project.

**Comment by tmontaigu** [comment]
When I openned this issue i was planning on using [milksnake](https://github.com/getsentry/milksnake) + cffi  because it was the one I sucessfully used  out of the 3 options (rust-cpython, pyo3, cffi + milksnake) 2 year ago.

I remember that passing errors from the rust side to python was kinda unpleasant using cffi so 'll re-evaluate what i'm gonna use by looking at how pyo3 and rust-cpython evolved.

But what's seems nice with [maturin](https://github.com/PyO3/maturin) is that it seems to work with
pyo3, rust-cpython and cffi to create python packages so thanks for suggesting it.

Currently the LAZ versions implemented works (or at least they work on the LAZ data I tested it on) so i'll try to tackle this issue
